analytics-core/src/main/scala/org/ekstep/analytics/framework/util/MergeUtil.scala

```diff
@@ -74,8 +74,29 @@ class MergeUtil {
           columnOrder = columnOrder.filter(col=> reportDfColumns.contains(col))
           MergeResult(mergeReport(rollupCol,rollupFormat,deltaDF, reportDF, mergeConfig, mergeConfig.merge.dims), reportDF,
             StorageConfig(storageType, postContainer, path.getParent.getName,Option(storageKey),Option(storageSecret)))
+        case "s3" =>
+          var deltaDF = fetchOSSFile(filePaths("deltaPath"),
+            mergeConfig.deltaFileAccess.getOrElse(true), mergeConfig.container)
+            deltaDF = (if(deltaDF.columns.contains("Date")) deltaDF.withColumn(rollupCol,
+              date_format(col("Date"), rollupFormat)) else deltaDF).dropDuplicates()
+          val reportFile = fetchOSSFile(filePaths("reportPath"), mergeConfig.reportFileAccess.getOrElse(true),
+            postContainer)
+          val reportDF = if (null == reportFile) {
+            sqlContext.createDataFrame(sc.emptyRDD[Row], if (null == reportSchema) {
```

Comment:
reportSchema is always initialised to null. What is the use of this if-else statement?

Reply:
Same as the previous comment, this is just the replica of the case for azure ( line 64 ). We just replicated the existing working logic of azure to make it work for s3 type of storage